sqlglot/optimizer/qualify_columns.py

```diff
@@ -122,6 +123,30 @@ def _expand_using(scope, schema):
                 scope.replace(column, replacement)
 
 
+def _expand_group_numbers(scope):
+    selects = scope.expression.args.get("expressions")
+    group = scope.expression.args.get("group")
+    if not group:
+        return
+
+    new_expressions = []
+    for expression in group.args.get("expressions"):
+        if isinstance(expression, exp.Literal) and expression.is_int:
+            try:
+                select = selects[int(expression.this) - 1]
+            except IndexError:
+                # pylint: disable=raise-missing-from
+                raise OptimizeError(f"Unknown output column: {expression.name}")
+            if isinstance(select, exp.Alias):
+                select = select.this
+            new_expressions.append(select.copy())
```

Comment:
what's this copy for?

Reply:
we need to copy the expression, or else we'll have one object with 2 parents (well, 1 parent, but I think you get the point)